### ISSUE TITLE
[Upgrade] Not possible to upgrade to prod version using docker img

### ISSUE THREAD

**Issue by avano** [syndesisio/syndesis#2577]
## This is a...

<pre><code>
[ ] Feature request
[ ] Regression (a behavior that used to work and stopped working in a new release)
[X] Bug report  
[ ] Documentation issue or request
</code></pre>

## The problem

Currently it is not possible to upgrade to newer version using docker container, because of preflight check failing. The version defined the template in prod version upgrade container specifies syndesis: "1.3" version.

The upgrade script then fails here:

```
    if [ -n "${SYNDESIS_VERSION:-}" ]; then
        if [ "$SYNDESIS_VERSION" != "${target_version}" ]; then
            echo "Internal error: Container template's version is not the same as upgrade container tag"
            echo "- Container version:               $SYNDESIS_VERSION"
            echo "- Version extracted from template: $target_version"
            exit 1
        fi
    fi 
```

Because the container variable SYNDESIS_VERSION is set to "1.3.5.fuse-000002-redhat-1" and the "target_version" is parsed from the template and therefore resolves to "1.3" and the if shuts everything down

## Tasks involved / Steps to Reproduce

1.
2.
3.
4.

**Comment by avano** [comment]
@paoloantinori @rhuss @dsimansk FYI

**Comment by paoloantinori** [comment]
I had lost visibility of this one. Is this still  a blocker?

**Comment by avano** [comment]
Hi,

Darren fixed this in the prod upgrade image, so not a blocker anymore, but still an issue. The question is if the syndesis template should specify syndesis version "1.3" instead of "1.3.6.xyz-redhat-1". If yes, then the preflight check should be reworked to take that into account.

**Comment by rhuss** [comment]
Sorry, I'm still trying to catch up. Where does this 1.3.6.xyz-redhat-1 comes from ? This is definitely not an image tag which should be referenced in any template, but a Maven like artefact version.

**Comment by avano** [comment]
Hi Roland,

the problem probably is that you can understand SYNDESIS_VERSION as two things in a productized build - "1.3" and "1.3.6.xyz-redhat-1". That is not a problem in upstream where tag and maven version are the same.

Currently Darren sets the SYNDESIS_VERSION in upgrade container to "1.3.6.xyz-redhat-1" and that causes this issue. The question is, what is the correct string there :-)

**Comment by dsimansk** [comment]
@rhuss @avano @paoloantinori 
In upstream release we set one version, e.g. `1.3.6`. That is used for Maven artifacts versioning and also image tags. Therefor Maven `syndesis.version == syndesis.image.tag`.

However, in the productized image tags has different versioning scheme given by build system, e.g. `1.0-1`. Translated as release version `1.0`, first build `-1`. And Maven version in as `1.3.6.fuse-000001-redhat-1`.

The above creates confusion what is actually `SYNDESIS_VERSION` correct value. Imho as Roland stated env variable `SYNDESIS_VERSION` is an image tag therefor value in productized upgrade image should be `1.0-1` from example.

There's another a little twist though. Current build and release process is also moving `1.0` to latest build. E.g. if you have set of respins `[1.0-1, 1.0-2, 1.0-3]`, there'll be always `1.0` => `1.0-3`.

The safest option here seems to be 'fuse-ignite-upgrade' env var `SYNDESIS_VERSION=1.0` and then upgrade template can be initialized with 1.0 version. Next release will be probably tagged `1.1` etc.

**Comment by dsimansk** [comment]
@rhuss could we sort out that value for `SYNDESIS_VERSION` variable. So @djcoleman could set it up during prod build.

**Comment by rhuss** [comment]
@dsimansk you are absolutely right, the "versions" we are talking about here are always image tags (so I probably have should called it 'tag', sorry for that)

So, the version to provide to "build.sh" as argument when creating the upgrade image should be the Docker image tag to which to upgrade which is the same tag as used for the syndesis-upgrade image.

So, considering its about to create an update image for updating to 1.0-3 the image should be build in upstream with 

```
build.sh 1.0-3
```

The upgrade image include also the template with the target version taken from [here](https://github.com/syndesisio/fuse-ignite-install/blob/master/resources/fuse-ignite-oso.yml#L183).

But as I look at it, I also see an issue here with the templates always specifying the 'minor version' (without patch-level, e.g. 1.0), so the preflight check probably should also only check up to the minor version. 

So I will update the check for this.

tl;dr - don't use Maven like release version numbers for the upgrade scripts, it's all about the image tags.

**Comment by paoloantinori** [comment]
@rhuss can we close this?

**Comment by rhuss** [comment]
Hmm, not sure whether this issue went below my radar. I have to verify still, whether I added the proposed fix already or not.
